### ISSUE TITLE
Git clone is failing when blobs filter is unavailable on platform

### ISSUE THREAD

**Issue by zharinov** [renovatebot/renovate#13440]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### Please select which platform you are using if self-hosting.

Gitea

### If you're self-hosting Renovate, tell us what version of the platform you run.

Latest

### Describe the bug

Gitea doesn't support shallow clone yet, which causes our our HTTPS authenticated clone to fail.

```sh
❯ git clone --filter=blob:none https://try.gitea.io/zharinov/shallow-test.git
Cloning into 'shallow-test'...
fatal: error reading section header 'shallow-info'

❯ git clone --filter=blob:none git@try.gitea.io:zharinov/shallow-test.git
Cloning into 'shallow-test'...
warning: filtering not recognized by server, ignoring
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (3/3), done.
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Performing blobless clone (repository=zharinov/shallow-test)
DEBUG: git clone error (repository=zharinov/shallow-test)
       "err": {
         "task": {
           "commands": [
             "clone",
             "--filter=blob:none",
             "https://**redacted**@try.gitea.io/zharinov/shallow-test.git",
             "."
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "Cloning into '.'...\nfatal: error reading section header 'shallow-info'\n",
         "stack": "Error: Cloning into '.'...\nfatal: error reading section header 'shallow-info'\n\n    at Object.action (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)"
       }
 WARN: Host error (repository=zharinov/shallow-test)
       "hostType": "git",
       "lookupName": undefined,
       "err": {
         "task": {
           "commands": [
             "clone",
             "--filter=blob:none",
             "https://**redacted**@try.gitea.io/zharinov/shallow-test.git",
             "."
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "Cloning into '.'...\nfatal: error reading section header 'shallow-info'\n",
         "stack": "Error: Cloning into '.'...\nfatal: error reading section header 'shallow-info'\n\n    at Object.action (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/sergio/Work/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)"
       }
 INFO: External host error causing abort - skipping (repository=zharinov/shallow-test)
DEBUG: Unknown res (repository=zharinov/shallow-test)
       "res": "external-host-error"
```

</details>

### Have you created a minimal reproduction repository?

https://try.gitea.io/zharinov/shallow-test

**Comment by viceice** [comment]
You can enable it in gitea.

**Comment by viceice** [comment]
https://docs.gitea.io/en-us/clone-filters/

**Comment by zharinov** [comment]
Thank you, now I see. So, maybe log this link for this particular stack trace?

**Comment by viceice** [comment]
You need this in `.gitconfig`:

```
[uploadpack]
        allowfilter = true
```

Are you using the official docker image?

**Comment by zharinov** [comment]
No, I've just tried to run with their public test instance.

**Comment by viceice** [comment]
🤔 For me it works even without the upload filter, but it will do a full clone then automatically. Maybe they have some extra filtering on the test instance?

**Comment by zharinov** [comment]
Possible ways to fix are:
- Make `fullClone` configurable by the Renovate admin
- Graceful fallback to full cloning

**Comment by bprins** [comment]
> 🤔 For me it works even without the upload filter, but it will do a full clone then automatically. Maybe they have some extra filtering on the test instance?

I'm having the same issue as @zharinov using the gitea official docker image (git version 2.30.2) without explicitly disabling clone filters (so enabled by default). Adding `allowfilter = true` to the my gitea `.gitconfig` `[uploadpack]` section fixed it.

**Comment by proton-ab** [comment]
This might've changed on Gitea side very recently. v1.15 should work as @viceice mentioned (no configuration necessary) while v1.16-rc1 requires explicitly enabling `allowfilter = true` in `.gitconfig`. Interestingly this has worked in v1.16 before rc1, so the change must've happened very recently (last 3 weeks or so of v1.16 development cycle before feature-freeze for rc1)

**Comment by viceice** [comment]
That pr explicit enables filters, so it should just work.

Are you using the official docker image? As the filter only work with git 2.22+. See PR changes.

**Comment by proton-ab** [comment]
>Are you using the official docker image? As the filter only work with git 2.22+. See PR changes.

Using `1.16.0-rc1-rootless` for Gitea and custom image based on `renovate/renovate:31-slim` with PHP, composer and golang built-in (updated daily).

Both meet requirements:
```sh
# docker run -ti ***/renovate sh
$ git --version
git version 2.35.0

# docker exec -ti gitea sh
$ git --version
git version 2.30.2
```

>That pr explicit enables filters, so it should just work.

Indeed, it should - yet it doesn't, unless
```ini
[uploadpack]
  allowfilter = true
```
is added to `.gitconfig`.

---

<details><summary>Root cause analysis</summary>
<p>

It used to work on previous development snapshot of v1.16 and I can pinpoint it to specific and and hour:
```
-rw-r--r-- 1 renovate renovate 473784 Jan 27 03:04 renovate-270120220300.log
-rw-r--r-- 1 renovate renovate  34663 Jan 27 04:00 renovate-270120220400.log
```

The time matches when Gitea was upgraded to `rc1` - during that time it has executed migrations 205,206,207,208 which means that the issue was introduced on Gitea between [6fe756dc93f3704079d942f0a29966b61aed8aec](https://github.com/go-gitea/gitea/commit/6fe756dc93f3704079d942f0a29966b61aed8aec) and [623c93ff46798e86636dba24f5a48bb35e12b9a5](https://github.com/go-gitea/gitea/commit/623c93ff46798e86636dba24f5a48bb35e12b9a5). Sadly Gitea does not log its own version when starting up so I can not pinpoint the exact commit that it has been upgraded from. 

Immediately after upgrade it started failing for these with:
```
2022/01/27 04:00:09 ...ers/web/repo/http.go:502:serviceRPC() [E] Fail to serve RPC(upload-pack) in /var/lib/repos/***/***.git: exit status 128 - fatal: unexpected line: 'filter blob:none'
```

As such, I believe that the issue has been indirectly introduced by previously mentioned commit - it tries to enable the functionality by default, however by doing so it trips on by failing to properly construct valid parameters for git command as explained on https://github.com/go-gitea/gitea/pull/18373
</p>
</details>

---

Unsure at the moment if there's anything that Renovate should do on its side. This will be fixed soon on Gitea side and the fix will be included in official 1.16 release version. Renovate is trying its best to act as it should and as far as I understand the issue at the moment, it's a server (Gitea in this case) that is acting incorrectly by failing a valid request.

**Comment by viceice** [comment]
I didn't see any issues with official v1.16 docker image. (Rootless)

**Comment by proton-ab** [comment]
Correct @viceice, https://github.com/go-gitea/gitea/commit/42991dc89a7f3dadb7ad3b36d18b1a74628ec272 is commit that fixes it for 1.16 branch (backport of https://github.com/go-gitea/gitea/pull/18373)
